### PR TITLE
Add custom unsubscribe url option in a campaign

### PR DESCRIPTION
**Pull request by raphaelhittich**
https://github.com/Mailtrain-org/mailtrain/pull/451 should be pulled first, otherwise there could be conflicts.

### CODE REVIEW

**Comment by bures** [comment]
Thanks.

File: services/sender.js [hunk 1]
```diff
@@ -381,7 +381,7 @@ function formatMessage(message, callback) {
 
                                 let listUnsubscribe = null;
                                 if (!list.listunsubscribeDisabled) {
-                                    listUnsubscribe = url.resolve(configItems.serviceUrl, '/subscription/' + list.cid + '/unsubscribe/' + message.subscription.cid);
+                                    listUnsubscribe = campaign.unsubscribe ? tools.formatMessage(configItems.serviceUrl, campaign, list, message.subscription, campaign.unsubscribe) : url.resolve(configItems.serviceUrl, '/subscription/' + list.cid + '/unsubscribe/' + message.subscription.cid);
```
Comment (njam) [inline services/sender.js]: @raphaelhittich does this mean the "unsubscribe" setting can contain merge tags?

Comment (raphaelhittich) [reply services/sender.js]: Exactly, for example:

https://example.com/unsubscribe/[MERGE_ID]

Comment (njam) [reply services/sender.js]: Nice!

**Pull request merged**